src/components/TodoInfo.tsx

```diff
@@ -0,0 +1,221 @@
+import classNames from 'classnames';
+import { useEffect, useRef, useState } from 'react';
+import { Todo } from '../types/Todo';
+import { deleteTodo, updateTodo } from '../api/todos';
+import { useTodoContext } from '../context';
+import { Errors } from '../types/Errors';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoInfo = ({ todo }: Props) => {
+  const [isEditing, setIsEditing] = useState<number | null>(null);
```

Comment:
when i see ;isEditing i feel like it should be boolean value. You can do that or change name to editTodoId